proxies/nodejs/handlers/location.ts

```diff
@@ -0,0 +1,153 @@
+        const resultData = await invokeCommand(
+            entity,
+            command,
+            params,
+            body.isAsync
+        )
+
+        const entityType = getEntityFromType(resultData.constructor.name)
+
+        const commandId = data.commandResults[entityType.toLowerCase()] !== undefined ?
+            Object.keys(data.commandResults[entityType.toLowerCase()]).length :
+            0
+
+        if (data.commandResults[entityType.toLowerCase()] === undefined) {
+            data.commandResults[entityType.toLowerCase()] = {}
+        }
+        data.commandResults[entityType.toLowerCase()][commandId] = resultData
+
+        ctx.status = 200
+        ctx.set('Location', `command/${entityType.toLowerCase()}/${commandId}`)
+        ctx.body = {
+            entityType: entityType,
+            data: resultData,
+            logs: [], // TODO add logs here
+        }
+        console.log('dataObject: ', data)
+        console.log('logger: ', entity.logger.info)
+
+    } catch (error) {
+        console.error(error)
+        if (body.isAsync) {
+            ctx.status = 200
+            ctx.body = {
+                asyncError: error.message,
+                errorCode: error.code,
+            }
+        } else {
+            ctx.status = 200
+            ctx.body = {
+                errorCode: error.code,
+                exception: error.message,
+            }
+
+        }
+    }
+}
+
+const getEntityFromLocation = (location: string, data: Data) => {
+    const urlParts = location.split('/')
+
+    /**
+     * location string is in the form of:
+     *  - URL = /entitiyType/id
+     *  - body params = entityType/id
+     * and therefore split on `/` will return an array of length 3 for URL and 2 for body params
+     */
+    if (urlParts.length === 3 || urlParts.length === 2) {
+        const entityType = urlParts[urlParts.length - 2]
+        const id = urlParts[urlParts.length - 1]
+        let entity
+        if (entityType === 'user') {
+            entity = data.users[id]
+        }
+        if (entityType === 'client') {
+            entity = data.clients[id]
+        }
+        return entity
+    } else if (urlParts.length === 4) {
+        const command = urlParts[urlParts.length - 2]
+        const entityType = urlParts[urlParts.length - 2]
+        const id = urlParts[urlParts.length - 1]
+        return data[command][entityType][id]
+    }
+    return undefined
+}
+
+const parseParams = (params: object | any, data: Data): parsedParams => {
+    const parsedParams: parsedParams = []
+    params.forEach((element) => {
+        if (element.value !== undefined) {
+            parsedParams.push(element.value)
+        } else if (element.location !== undefined) {
+            parsedParams.push(getEntityFromLocation(element.location, data))
+        }
+    })
+    return parsedParams
+}
+
+const invokeCommand = async (
+    entity: DVCClient | DVCUser | DVCVariable | any,
```

Comment:
once we have typescript linting we should remove any params that have `any` type, we should know explicitly what types a function can have